### PR TITLE
sdr-radio: APT scan-line image buffer (closes #478)

### LINKED ISSUE #478: APT scan-line assembly + 2D image buffer
## Scope

Consume the stream of synchronized scan lines from the APT decoder (#477) and assemble them into a growing 2D grayscale image. Handle pass-start, late-joining passes, dropped lines, and pass-end gracefully.

## Responsibilities

- Maintain an image buffer of width 2080 (APT line width) and variable height (grows with pass duration)
- Add new lines as they arrive
- Handle sync loss: gap-fill with black (or last-good line) if a line fails sync detection
- Track pass boundaries: start fresh on a new satellite pass, finalize image at pass end
- Provide a snapshot API — callers can grab the current partial image at any time (for live display)

## Data model

```rust
pub struct AptImage {
    width: usize,  // 2080
    lines: Vec<AptImageLine>,
    pass_start: Instant,
    channel_a_id: Option<AvhrrChannel>,  // filled from telemetry (ticket #TBD)
    channel_b_id: Option<AvhrrChannel>,
}

pub struct AptImageLine {
    pixels: [u8; 2080],
    sync_quality: f32,
    received_at: Instant,
}
```

## Implementation notes

- No allocation in the per-line hot path — pre-size the `Vec<AptImageLine>` from expected pass duration (15 min × 2 lines/sec = 1800 lines max)
- Separate concern from DSP — this is buffer/state logic, unit-testable with synthetic line inputs
- Snapshot returns a cheap reference, not a clone — caller can save to PNG elsewhere

## Out of scope
- PNG encoding (lives in the UI viewer ticket)
- Telemetry wedge calibration

Depends on: #477

Part of epic #468.

### PR DESCRIPTION
**Pull request by jasonherald**
## Summary

Second sub-ticket of the NOAA APT epic (#468). Pure data structure that sits one stage downstream of the DSP decoder shipped in PR #492 — takes the stream of `AptLine`s the decoder emits and stitches them into a growing 2D greyscale image.

\`\`\`text
    AptDecoder ──[AptLine]──▶  AptImage::push_line  ──▶  AptImage::lines()
                                                         │
                                                         ▼
                                             live viewer / PNG export
\`\`\`

## What's in the box

- \`AptImage\` — owning buffer for one satellite pass. Pre-reserves \`DEFAULT_MAX_LINES = 1800\` (15-min pass × 2 lines/sec) so the typical pass is alloc-free.
- \`AptImageLine\` — pixels + sync_quality + received_at.
- \`MIN_VALID_SYNC_QUALITY = 0.5\` threshold; sub-threshold lines gap-fill to black so flapping squelch / sync-loss reads as a clean dark band, not a noise strip. Quality score is preserved verbatim either way so a renderer can still surface a quality overlay.
- \`AvhrrChannel\` enum + setters on the image — slot for wedge-16 telemetry decode (#479) to fill in.

## What's not in the box

- No wiring into the live radio path yet — that's #482 (auto-record on overhead pass) and #483 (live viewer + PNG export).
- No telemetry-wedge decode — that's #479. Channel IDs stay \`None\` until then.

## Test plan

- [x] 9 new unit tests in \`sdr_radio::apt_image::tests\`: empty image, high-quality preservation, low-quality gap-fill (pixels black, quality kept), boundary case at exactly the threshold (\`>=\`), capacity invariance up to reservation, insertion ordering + timestamp monotonicity, channel-id round-trip, pass_start round-trip, and \`WIDTH == LINE_PIXELS == 2080\`.
- [x] \`cargo test -p sdr-radio\` — all 94 tests pass (85 existing + 9 new).
- [x] \`cargo clippy --all-targets --workspace -- -D warnings\` clean.
- [x] \`cargo fmt --all -- --check\` clean.
- [x] \`cargo build --workspace\` clean (post-merge with main / Mac UI #493).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added NOAA APT satellite image assembly: build complete passes from incoming scan-lines with per-line timestamps, channel A/B IDs, sync-quality tracking and automatic gap-filling for low-quality segments. Exposes pass metadata, snapshot access to accumulated lines, capacity-aware accumulation, and stable ordering of inserted lines.

### CODE REVIEW

File: crates/sdr-radio/src/apt_image.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    /// Build an `AptLine` with the given quality and a deterministic
+    /// non-zero pixel pattern so we can verify pixel preservation.
+    fn synth_line(quality: f32) -> AptLine {
+        let mut line = AptLine {
+            sync_quality: quality,
+            ..AptLine::default()
+        };
+        for (i, p) in line.pixels.iter_mut().enumerate() {
+            *p = (i % 251) as u8; // any non-trivial pattern
+        }
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply crates/sdr-radio/src/apt_image.rs]: Fixed in 1b73c0a. Hoisted all four:

* `TEST_PIXEL_PATTERN_MODULUS = 251` — prime under 256 for the synthetic pixel ramp.
* `TEST_ORDERED_INSERT_COUNT = 5` + `TEST_ORDERED_STEP_MS = 500` — ordering test sample count and cadence (matches the real APT line rate).
* `EXPECTED_APT_WIDTH_PIXELS = 2080` — pinned *independently* of `LINE_PIXELS` so any future protocol-width drift in `sdr-dsp` gets caught loudly here instead of silently propagating.

Each constant has a one-line rationale comment explaining why the value was chosen, matching the pattern established in `sdr_dsp::apt::tests`.

File: crates/sdr-radio/src/apt_image.rs [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    /// [`AptImage::with_capacity`] if you want to tune the upper bound
+    /// (e.g. for a higher-elevation pass or for tests).
+    #[must_use]
+    pub fn new(pass_start: Instant) -> Self {
+        Self::with_capacity(pass_start, DEFAULT_MAX_LINES)
+    }
+
+    /// Start a fresh pass, pre-reserving room for `max_lines` scan lines.
+    /// Pushing past `max_lines` still works but pays one Vec reallocation.
+    #[must_use]
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply crates/sdr-radio/src/apt_image.rs:117]: Fixed in 7005027. You're right — Vec's growth policy doubles each time, so pushing far past `max_lines` can realloc repeatedly. Updated the doc to "one or more Vec reallocations as it doubles to fit", which matches actual behaviour.
